src/resolvers.js

```diff
@@ -276,7 +277,10 @@ export class NewInstance {
   * @return Returns the matching instance from the parent container
   */
   get(container) {
-    const instance = container.invoke(this.key);
+    let dynamicDependencies = this.dynamicDependencies.length > 0 ?
+      this.dynamicDependencies.map(dependency => dependency["protocol:aurelia:resolver"] ?
+        dependency.get(container) : container.invoke(dependency)) : undefined;
```

Comment:
@EisenbergEffect I've changed `invoke` to `get` and also added more test for the resolver and the decorator, plus enforce existing test of `NewInstance` to check the instance really being a new one.